src/wallet/storage.rs

```diff
@@ -157,3 +157,103 @@ impl FileData {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use bip39::Mnemonic;
+    use bitcoin::Network;
+    use bitcoind::tempfile::tempdir;
+
+    #[test]
+    fn test_write_and_read_wallet_to_disk() {
+        let temp_dir = tempdir().unwrap();
+        let file_path = temp_dir.path().join("test_wallet.json");
+        let mnemonic = Mnemonic::generate(12).unwrap().to_string();
+
+        let original_wallet_store = WalletStore::init(
+            "test_wallet".to_string(),
+            &file_path,
+            Network::Bitcoin,
+            mnemonic.clone(),
+            "passphrase".to_string(),
+        )
+        .unwrap();
+
+        original_wallet_store.write_to_disk(&file_path).unwrap();
+
+        let read_wallet = WalletStore::read_from_disk(&file_path).unwrap();
+
+        assert_eq!(read_wallet.wallet_name, original_wallet_store.wallet_name);
+        assert_eq!(read_wallet.network, original_wallet_store.network);
+        assert_eq!(
+            read_wallet.external_index,
+            original_wallet_store.external_index
+        );
+        assert_eq!(
+            read_wallet.incoming_swapcoins,
+            original_wallet_store.incoming_swapcoins
+        );
+        assert_eq!(
+            read_wallet.outgoing_swapcoins,
+            original_wallet_store.outgoing_swapcoins
+        );
+        assert_eq!(
+            read_wallet.prevout_to_contract_map,
+            original_wallet_store.prevout_to_contract_map
+        );
+        assert_eq!(
+            read_wallet.fidelity_scripts,
+            original_wallet_store.fidelity_scripts
+        );
+        assert_eq!(
+            read_wallet.offer_maxsize,
+            original_wallet_store.offer_maxsize
+        );
+        assert_eq!(read_wallet.master_key, original_wallet_store.master_key);
+    }
+
+    #[test]
+    fn test_file_data_save_and_load() {
```

Comment:
Actually, an `WalletStore` read/write implies `FileData` read/write. If I am not missing anything, then this is already covered so just one test for this module is enough.